sympy/matrices/dense.py

```diff
@@ -313,10 +315,21 @@ def _new(cls, *args, copy=True, **kwargs):
         else:
             rows, cols, flat_list = cls._handle_creation_inputs(*args, **kwargs)
             flat_list = list(flat_list) # create a shallow copy
+
+            types = set(map(type, flat_list))
+            if not all(issubclass(typ, Expr) for typ in types):
+                SymPyDeprecationWarning(
+                    feature="non-Expr objects in a Matrix",
+                    useinstead="list of lists or some other data structure",
```

Comment:
perhaps we could advertise `TableForm` here; and maybe slicing of TableForm could be added.

Reply:
That's a good point. I've mentioned TableForm in the message